app/ide-desktop/lib/content/esbuild-config.ts

```diff
@@ -106,27 +78,82 @@ export async function* filesToCopyProvider(wasmArtifacts: string, assetsPath: st
  * Generate the builder options.
  */
 export function bundlerOptions(args: Arguments) {
-    const { outputPath, ensoglAppPath, wasmArtifacts, assetsPath } = args
+    const { outputPath, ensoglAppPath, wasmArtifacts, assetsPath, devMode } = args
     const buildOptions = {
         // Disabling naming convention because these are third-party options.
         /* eslint-disable @typescript-eslint/naming-convention */
         absWorkingDir: THIS_PATH,
         bundle: true,
-        entryPoints: [path.resolve(THIS_PATH, 'src', 'index.ts')],
+        loader: {
+            '.html': 'copy',
+            '.css': 'copy',
+            '.wasm': 'copy',
+            '.svg': 'copy',
+            '.png': 'copy',
+            '.ttf': 'copy',
+        },
+        entryPoints: [
+            pathModule.resolve(THIS_PATH, 'src', 'index.ts'),
+            pathModule.resolve(THIS_PATH, 'src', 'index.html'),
+            pathModule.resolve(THIS_PATH, 'src', 'run.js'),
+            pathModule.resolve(THIS_PATH, 'src', 'style.css'),
+            pathModule.resolve(THIS_PATH, 'src', 'docsStyle.css'),
+            ...wasmArtifacts.split(pathModule.delimiter),
+            ...fsSync
+                .readdirSync(assetsPath)
+                .map(fileName => pathModule.resolve(assetsPath, fileName)),
+        ].map(path => ({ in: path, out: pathModule.basename(path, pathModule.extname(path)) })),
         outdir: outputPath,
         outbase: 'src',
         plugins: [
+            {
+                // This is a workaround that is needed because esbuild panics when using `loader: { '.js': 'copy' }`.
+                name: 'pkg-js-is-cjs',
+                setup: build => {
+                    build.onLoad({ filter: /\/pkg.js$/ }, async ({ path }) => ({
+                        contents: await fs.readFile(path),
+                        loader: 'copy',
+                    }))
+                },
+            },
+            {
```

Comment:
I'm not sure about inlining such things. While it is not necessary to keep this as a separate package, it should be extracted, documented and be reusable, e.g. if "client" bundler ever wants to copy directories.